torch/quantization/quantize.py

```diff
@@ -70,7 +70,10 @@ def add_observer(module, skip_list=DEFAULT_SKIP_LIST):
             forward_hooks
     """
     for child in module.children():
-        add_observer(child)
+        if type(child) == nnq.FloatFunctional:
```

Comment:
This code should be called once you know you are at a leaf module and can check for qconfig:
Remove this change here and insert below:
```python
 # Insert observers only for leaf nodes, note that this observer is for
    # the output of the module, for input QuantStub will observe them
    if hasattr(module, 'qconfig') and module.qconfig is not None and \
       len(module._modules) == 0 and type(module) not in skip_list:
        # observer and hook will be gone after we swap the module
        if type(module) == nnq.FloatFunctional:
              module.observer = module.qconfig.activation()
      else:
           module.add_module('observer', module.qconfig.activation())
           module.register_forward_hook(_observer_forward_hook)
```

Reply:
This is already there later in the code. At the same time, FloatFunctional is not expected to have children, so it will always be a child. The reason is that the FloatFunctional will raise an error if `forward` is called, which means it's always a leaf. At the same time, because this this is a recursive function, semantically it is the same

Reply:
Made the change anyway :)